### PR TITLE
Improve channel getters on events

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This adds all the remaining getters for channel related events. I intentionally did not update any interaction events because that will be addressed by the rework there.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/events/channel/GenericChannelEvent.java [hunk 1]
```diff
@@ -45,30 +56,49 @@ public boolean isFromGuild()
...
     {
-        return this.channel;
+        return getChannelType().isAudio();
     }
 
     /**
      * The {@link net.dv8tion.jda.api.entities.Guild Guild} in which this channel event happened.
-     * <br>If this channel event was not received in a {@link net.dv8tion.jda.api.entities.TextChannel TextChannel},
+     * <br>If this channel event was not received in a {@link TextChannel TextChannel},
      * this will throw an {@link java.lang.IllegalStateException}.
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/events/channel/GenericChannelEvent.java:98]:
```suggestion
```

File: src/main/java/net/dv8tion/jda/api/events/channel/GenericChannelEvent.java [hunk 2]
```diff
@@ -83,5 +113,184 @@ public Guild getGuild()
...
+    @Nonnull
+    public Channel getChannel()
+    {
+        return this.channel;
+    }
+
+    /**
+     * The {@link GuildChannel} this event was received for.
+     * <br>If this event was not received from a {@link net.dv8tion.jda.api.entities.Guild Guild},
+     * this will throw an {@link java.lang.IllegalStateException}.
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/events/channel/GenericChannelEvent.java:125]:
```suggestion
```

File: src/main/java/net/dv8tion/jda/api/events/channel/GenericChannelEvent.java [hunk 3]
```diff
@@ -83,5 +113,184 @@ public Guild getGuild()
...
+    @Nonnull
+    public MessageChannel getMessageChannel()
+    {
+        if (!getChannelType().isMessage())
+            throw new IllegalStateException("This event did not happen for a message channel");
+        return (MessageChannel) channel;
+    }
+
+    @Nonnull
+    public GuildMessageChannel getGuildMessageChannel()
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/api/events/channel/GenericChannelEvent.java:153]: Docs for both

**Comment by MinnDevelopment** [comment]
Closing in favor of #2138

**Pull request closed without merging**